avatars/client.py

```diff
@@ -123,4 +122,28 @@ def request(
         if result.headers["content-type"] == "application/json":
             return result.json()
         else:
-            return result.content
+            return result.text
+
+    def _get_file_argument(
+        self, file: Optional[Union[StringIO, BytesIO]]
+    ) -> Optional[Dict[str, Tuple[str, bytes, str]]]:
+
+        if not file:
+            return
+
+        filename = str(Path(file.name).name) if hasattr(file, "name") else "file.csv"
+
+        content = file.read(MAX_FILE_LENGTH)
+
+        if sys.getsizeof(content) >= MAX_FILE_LENGTH:
```

Comment:
It would be easier to check if there is still some unread content:

https://docs.python.org/3/library/io.html#io.RawIOBase.read